Allow to skip NVM installation in setup-local-env.sh

desaiuditd opened the pull request:
## Description
Related to #9120.

<!-- Please describe what you have changed or added -->
I'm using [nodenv](https://github.com/nodenv/nodenv/) as Node Version Manager. So it's conflicting with the NVM setup script in `./setup-local-env.sh`.

This PR introduces a flag option for the `setup-local-env.sh` script to skip NVM installation and setup, if needed. By default, it will behave as is.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
- Running `./bin/setup-local-env.sh --skip-nvm-setup` on the local machine will skip all the steps related to NVM, and directly install `npm` packages.
<!-- Include details of your testing environment, tests ran to see how -->
- So far, I've only tested this on my Mac environment. I've used [`getopts`](https://ss64.com/bash/getopts.html) which seems to be in-built command in bash. So hopefully, it also works in Windows/Linux. Happy to listen from other OS users.
<!-- your change affects other areas of the code, etc. -->
- I've also updated the Getting Started doc for dev contributors.

## Types of changes
<!-- What types of changes does your code introduce?  -->
- Local Environment Setup script.
- Documentation to let people know about the new flag to skip NVM installation.
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested (Only on MacOS so far. Need help in testing this on other OS).
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

desaiuditd committed:
Allow to skip NVM installation in setup-local-env.sh

desaiuditd commented:
Reviewers' Note: Hide whitespace changes while reviewing. It will help to understand the new change in better way.

desaiuditd commented on bin/bootstrap-env.sh:
```diff
@@ -10,3 +10,16 @@ DOCKER_COMPOSE_FILE_OPTIONS="-f docker-compose.yml"
+	case "${opt}" in
+		-)
+			case "${OPTARG}" in
+				skip-nvm-setup)
```
Open to suggestions on renaming this flag name.

gziolo commented:
This change breaks Travis jobs causing them to run forever, see:
https://travis-ci.com/WordPress/gutenberg/builds/116557184

![Screen Shot 2019-06-24 at 09 39 44](https://user-images.githubusercontent.com/699132/60000205-05389100-9664-11e9-931d-edc14275bca1.png)

desaiuditd committed:
Remove the bash command to shift processed flags in setup script.

This was not needed, as each script is sourced explicitly in the current shell. So it will take its own arguments.
This will also fix the Travis CI timeout.

desaiuditd committed:
Trigger Travis

desaiuditd commented:
First commit was indeed a genuine mistake from my side and the build was failing because of that.

But last 2 builds doesn't look like relevant failures. In first, an e2e test failed  and in second, a unit test failed.

I am not able to conclude a straight forward reason, why they would fail due to my changes, which are essentially targeted towards build tooling and local development.

Can someone have a second look? 👀

desaiuditd committed:
Merge branch 'master' into add/allow-to-skip-nvm-installation-in-local-env-setup

desaiuditd commented:
Merged latest master into my branch and now tests are passing without any changes from my end. Should be good to review now.

senadir commented:
cc @pento does this sound like something we might want to have?
I have concerns about possible problems if we don't have a way to check for the node version used

youknowriad commented:
nvm is not really a standard to check or install node versions, I'd personally just remove all nvm related scripts... and just check if the node version is not right, ask the person to install the right one with its preferred method.

pento commented:
Yah, NVM was an interesting experiment in trying to ensure the correct version of Node was installed, but I don't think it was a successful as I would've liked.

#17004 will remove all of these scripts, so I'm going to close this PR in favour of that one.
